pkg/security/api/types.go

```diff
@@ -0,0 +1,88 @@
+package api
+
+import (
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+)
+
+// PodSpecSubjectReview checks whether a particular user/SA tuple can create the PodSpec.
+type PodSpecSubjectReview struct {
+	unversioned.TypeMeta
+
+	// Spec defines specification for the PodSpecSubjectReview.
+	Spec PodSpecSubjectReviewSpec
+
+	// Status represents the current information/status for the PodSpecSubjectReview.
+	Status PodSpecSubjectReviewStatus
+}
+
+type PodSpecSubjectReviewSpec struct {
+	// PodSpec is the PodSpec to check.
+	PodSpec kapi.PodSpec
+
+	// User is the user you're testing for.
+	// If you specify "User" but not "Group", then is it interpreted as "What if User were not a member of any groups.
+	// If User and Groups are empty, then the check is performed using *only* the ServiceAccountName in the PodSpec.
+	User string
+
+	// Groups is the groups you're testing for.
+	Groups []string
+}
+
+// PodSpecSubjectReviewStatus contains information/status for PodSpecSubjectReview.
+type PodSpecSubjectReviewStatus struct {
+	// AllowedBy is a reference to the rule that allows the PodSpec.  A `nil`, indicates that it was denied.
+	AllowedBy *kapi.ObjectReference
+
+	// Reason is optional.  It indicates why a request was allowed or denied.
+	Reason string
+
+	// PodSpec is the PodSpec after the defaulting is applied.
+	PodSpec kapi.PodSpec
+}
+
+// PodSpecSelfSubjectReview checks whether this user/SA tuple can create the PodSpec.
+type PodSpecSelfSubjectReview struct {
+	unversioned.TypeMeta
+
+	// Spec defines specification the PodSpecSelfSubjectReview.
+	Spec PodSpecSelfSubjectReviewSpec
+
+	// Status represents the current information/status for the PodSpecSelfSubjectReview.
+	Status PodSpecSubjectReviewStatus
+}
+
+// PodSpecSelfSubjectReviewSpec contains specification for PodSpecSelfSubjectReview.
+type PodSpecSelfSubjectReviewSpec struct {
```

Comment:
Done according to @smarterclayton comment https://github.com/openshift/origin/pull/8199/files#r64410411